aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,303 @@
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=4)]
+
+The sample application will redirect requests from **/apache-mod-rules-redirect/(.\*)** to **/redirected?id=$1**. The response status code is 302 (Found).
+
+[!code[Main](url-rewriting/sample/ApacheModRewrite.txt)]
+
+Original Request: **http://localhost:5000/apache-mod-rules-redirect/1234**
+
+![Browser window with Developer Tools tracking the requests and responses](url-rewriting/_static/add_apache_mod_redirect.png)
+
+##### Supported server variables
+The middleware supports the following Apache mod_rewrite server variables:
+* CONN_REMOTE_ADDR
+* HTTP_ACCEPT
+* HTTP_CONNECTION
+* HTTP_COOKIE
+* HTTP_FORWARDED
+* HTTP_HOST
+* HTTP_REFERER
+* HTTP_USER_AGENT
+* HTTPS
+* IPV6
+* QUERY_STRING
+* REMOTE_ADDR
+* REMOTE_PORT
+* REQUEST_FILENAME
+* REQUEST_METHOD
+* REQUEST_SCHEME
+* REQUEST_URI
+* SCRIPT_FILENAME
+* SERVER_ADDR
+* SERVER_PORT
+* SERVER_PROTOCOL
+* TIME
+* TIME_DAY
+* TIME_HOUR
+* TIME_MIN
+* TIME_MON
+* TIME_SEC
+* TIME_WDAY
+* TIME_YEAR
+
+#### IIS URL Rewrite Module rules
+To use rules that would normally apply to the IIS URL Rewrite Module, use `AddIISUrlRewrite()`. The first parameter takes an `IFileProvider`, while the second parameter is the path to your XML rules file, which is **IISUrlRewrite.xml** in the sample application. You must make sure that the rules file is deployed with the application. For more information and examples of IIS URL Rewrite Module rules, see [Using Url Rewrite Module 2.0](https://www.iis.net/learn/extensions/url-rewrite-module/using-url-rewrite-module-20) and [URL Rewrite Module Configuration Reference](https://www.iis.net/learn/extensions/url-rewrite-module/url-rewrite-module-configuration-reference).
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=5)]
+
+The sample application will rewrite requests from **/iis-rules-rewrite/(.\*)** to **/rewritten?id=$1**. The response is sent to the client with a 200 (OK) status code.
+
+[!code-xml[Main](url-rewriting/sample/IISUrlRewrite.xml)]
+
+Original Request: **http://localhost:5000/iis-rules-rewrite/1234**
+
+![Browser window with Developer Tools tracking the request and response](url-rewriting/_static/add_iis_url_rewrite.png)
+
+If you have an active IIS Rewrite Module functioning on the server that you wish to disable for an application using the middleware, you can disable the IIS module with a change to your **web.config** file.
```

Comment:
It should not be necessary to disable the module if you avoid adding any rules for it in the web.config.